### PR TITLE
Pull pcre2 from conda-forge

### PR DESCRIPTION
**Pull request by dfornika**
Addresses #1.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ requirements:
     - libgit2
     - arpack
     - suitesparse
+    - pcre2
```
Comment (bgruening) [inline recipe/meta.yaml]: I think you need to bump the build number.

Comment (dfornika) [reply recipe/meta.yaml]: Of course! Thanks, I keep forgetting about that.

**Comment by jakirkham** [comment]
This might need to be pinned. Here's the [ABI report]( https://abi-laboratory.pro/tracker/timeline/pcre2/ ).

**Comment by dfornika** [comment]
Ok. I don't have any experience interpreting an ABI report, but I did take a look at the julia github repo and saw that for julia 0.5.2 they pull in pcre2-10.21:

https://github.com/JuliaLang/julia/blob/v0.5.2/deps/Versions.make

julia 0.6.0 uses pcre2-10.23:

https://github.com/JuliaLang/julia/blob/v0.6.0/deps/Versions.make

**Comment by jakirkham** [comment]
So we would need to package `pcre2` 10.23 if we want to use the same pinning.

**Comment by dfornika** [comment]
This is julia-0.5.2, so we'd want pcre2-10.21, wouldn't we?

**Comment by jakirkham** [comment]
My mistake. Fixed my comment above.

**Comment by dfornika** [comment]
Relevant excerpt from build log:

```
regex.jl

pcre.jl

show.jl

LoadError("sysimg.jl",152,LoadError("show.jl",1238,ErrorException("PCRE JIT error: bad JIT option")))
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[2]: *** [/feedstock_root/build_artefacts/julia_1504301911995/work/julia-0.5.2/usr/lib/julia/sys.o] Error 1
make[1]: *** [julia-sysimg-release] Error 2
make[1]: *** Waiting for unfinished jobs....
LoadError("sysimg.jl",152,LoadError("show.jl",1238,ErrorException("PCRE JIT error: bad JIT option")))
make[2]: *** [/feedstock_root/build_artefacts/julia_1504301911995/work/julia-0.5.2/usr/lib/julia/sys-debug.o] Error 1
make[1]: *** [julia-sysimg-debug] Error 2
make: *** [install] Error 2
```

`show.jl` surrounding line 1238 is:

```
1222 function show(io::IO, tv::TypeVar)
...

1236   if !in_env && lb !== Bottom
1237       if ub === Any
1238            write(io, tv.name)
1239            print(io, ">:")
1240           show_bound(io, lb)
1241       else
```

**Comment by jakirkham** [comment]
It may be that version of `pcre2` is coming from `defaults`, which could be causing some issues. Would add a new (non-`master`) branch, possibly `10.21`, to the `pcre2` feedstock for the legacy version and PR the old version to that branch. Once we have a release for the old version we can retry this.

**Comment by dfornika** [comment]
I'm going to close this pull request and open a new one with the same purpose. I should have made a specific branch on my repo for this feature instead of using master.

**Pull request closed without merging**